src/action_handle_hashtags.py

```diff
@@ -0,0 +1,139 @@
+        print("Enter own hashtag search")
+        tab_bar = device(resourceId='com.instagram.android:id/tab_bar', className='android.widget.LinearLayout')
+        search_button = tab_bar.child(index=1)
+
+        # Two clicks to reset tab content
+        search_button.click.wait()
+    else:
+        print("Press search")
+        tab_bar = device(resourceId='com.instagram.android:id/tab_bar', className='android.widget.LinearLayout')
+        search_button = tab_bar.child(index=1)
+
+        # Two clicks to reset tab content
+        search_button.click.wait()
+        search_button.click.wait()
+
+        print("Open hashtag #" + hashtag)
+        search_edit_text = device(resourceId='com.instagram.android:id/action_bar_search_edit_text',
+                                  className='android.widget.EditText')
+        search_edit_text.set_text(hashtag)
+
+        search_hashtag_button = device(resourceId='com.instagram.android:id/fixed_tabbar_view',
+                            className='android.widget.FrameLayout').child(index=2)
+        search_hashtag_button.click.wait()
+
+        device.wait.idle()
+        hashtag_view = device(resourceId='com.instagram.android:id/row_hashtag_textview_tag_name',
+                               className='android.widget.TextView',
+                               text="#" + hashtag)
+
+        if not hashtag_view.exists:
+            print_timeless(COLOR_FAIL + "Cannot find hashtag #" + hashtag + ", abort." + COLOR_ENDC)
+            return False
+
+        hashtag_view.click.wait()
+
+        print("Open #" + hashtag + " recent posts")
+        hashtag_type_recent = device(resourceId='com.instagram.android:id/tab_layout',
+                                  className='android.widget.LinearLayout').child(index=1)
+        hashtag_type_recent.click.wait()
+
+    return True
+
+def open_photo(device,index_ran):
+        # recycler_view has a className 'androidx.recyclerview.widget.RecyclerView' on modern Android versions and
+        # 'android.view.View' on Android 5.0.1 and probably earlier versions
+        recycler_view = device(resourceId='com.instagram.android:id/recycler_view',
+                                className='androidx.recyclerview.widget.RecyclerView')
+        item_view = recycler_view.child(index=index_ran)
+        item_view.click.wait()
+
+
+def _iterate_over_posts(device, interaction, storage, on_interaction):
+    #random up to 12 because min show on screen intially is 12 the max is 20
+    posts_per_screen = None
+    interactions_count = 0
+    iterations = randint(2,12)
+    while True:
+        print("Iterate over visible hashtag posts")
+        screen_iterated_posts = 0
+
+        for x in range(2,iterations):
+
+            open_photo(device,x) 
+            profile_check = device(resourceId='com.instagram.android:id/row_feed_photo_profile_imageview',
+                                    className='android.widget.ImageView')
+            profile_check.click.wait()

```

Comment:
Hmm crashed on this line after 4 minutes with
`uiautomator.JsonRPCError: JsonRPC Error code: -32002, Message: android.support.test.uiautomator.UiObjectNotFoundException: UiSelector[CLASS=android.widget.ImageView, RESOURCE_ID=com.instagram.android:id/row_feed_photo_profile_imageview]`

Reply:
Seems that the problem is that posts in this feed are sequentially numbered, so after we iterate over, say, 12 posts (last index 13), we have to start from the index 14 after scrolling. As I see you always start from index 2, don't you?

Reply:
I can double check but once you scroll it reindexes all the photos.  Let me try to reproduce this error.